lib/grape/validations/types/build_coercer.rb

```diff
@@ -54,8 +58,22 @@ def self.build_coercer(type, method = nil)
 
         # Virtus will infer coercion and validation rules
         # for many common ruby types.
-        Virtus::Attribute.build(conversion_type, converter_options)
+        to_cache(type, method, Virtus::Attribute.build(conversion_type, converter_options))
       end
+
+      def self.from_cache(type, method)
```

Comment:
Like this block approach, did something similar to the latest version. Also not sure if we need to lock the write access to the hash for thread-safety.

The non locking version of this method would be something like:
```
def self.cache_instance_no_lock(type, method, &_block)
  @__cache.fetch(cache_key(type, method)) do |key|
    @__cache[key] = yield
  end
end
```